data/bundle.go

```diff
@@ -102,10 +104,33 @@ type BundleCommand struct {
 	Executable      []string  `yaml:",omitempty,flow" json:"executable,omitempty"`
 	LongDescription string    `yaml:"long_description,omitempty" json:"long_description,omitempty"`
 	Name            string    `yaml:"-" json:"-"`
+	Triggers        []Trigger `yaml:"triggers,omitempty" json:"trigger,omitempty"`
 	Rules           []string  `yaml:",omitempty" json:"rules,omitempty"`
 	Templates       Templates `yaml:",omitempty" json:"templates,omitempty"`
 }
 
+// Trigger represents the configuration for a command trigger as defined
+// in the bundles/commands/triggers section of the config.
+type Trigger struct {
+	Match string `yaml:"match" json:"match"`
+}
+
+func (c *BundleCommand) MatchTrigger(ctx context.Context, message string) (bool, error) {
+	for _, trigger := range c.Triggers {
+		if c == nil || len(trigger.Match) == 0 {
+			return false, nil
+		}
+		re, err := regexp.Compile(trigger.Match)
```

Comment:
Regex compilation is expensive. I'm wondering if these patterns should be compiled on installation, which would allow them to be cached and also would allow compilation errors to be report when the bundle is installed instead of when a command is triggered (which, if I'm reading this correctly, would prevent any other commands from executing?)

Not a blocker for this PR: but something to consider.

Reply:
I'd agree with that. We could probably make it part of an in-memory cache for performance in general.